chore: add custom properties example

smbea opened the pull request:
Context: form-js@0.7.0 blog post

smbea committed:
chore: add custom properties example

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+  return properties && properties.externalData
+}
+
+function mockFecth(_, _) {
```
```suggestion
function mockFetch(_, _) {
```

nikku commented on custom-properties/index.html:
```diff
@@ -0,0 +1,89 @@
+      "schemaVersion": 1,
+      "exporter": {
+        "name": "form-js",
+        "version": "0.1.0"
```
```suggestion
        "version": "0.7.0"
```

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+  const {
+    properties
+  } = field;
+  return properties && properties.externalData
```
```suggestion
  return properties && properties.externalData;
```

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+      field.values = res.map(option => ({
+        value: option.id,
+        label: option.name
+      }))
```
```suggestion
      }));
```

nikku commented on custom-properties/index.html:
```diff
@@ -0,0 +1,89 @@
+          "id": "Field_3",
+          "key": "approvedBy",
+          "properties": {
+            "externalData": "lala"
```
Does `lala` make any sense? If not, make it a boolean?

smbea replied:
It doesn't. I just changed it to an example API endpoint

nikku reviewed (changes requested):
What is missing:

* README

Good start!

nikku commented:
CC @ingorichtsmeier

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+  return properties && properties.externalData;
+}
+
+function mockFetch(_, _) {
```
```suggestion
async function preloadData(url, method) {
```

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+});
+
+// (2) populate form fields with external data
+loadForm.then((form) => {
```
```suggestion
loadForm.then(async (form) => {
```

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+    components
+  } = schema;
+
+  components.forEach(field => {
```
make a for of loop (so nested `await` works).

smbea replied:
`for` loop instead of `forEach`? May I ask why? -> Got it

nikku commented on custom-properties/customProperties.js:
```diff
@@ -0,0 +1,69 @@
+
+      // (2.1) use custom property "externalData" to get external API endpoint
+      const endpoint = properties.externalData;
+      const res = mockFetch(endpoint, "GET");
```
```suggestion
      const res = await preloadData(endpoint, "GET");
```

nikku reviewed (commented):
Suggestion to make the example more robust:

* add argument names to fetch
* make fetch async (this is what fetch is always going to be)

MaxTru commented:
Looks great!

* I tried it out and realized that nothing happens when clicking on `submit`. While this of course is not the focus of the example, I still think that this click on `submit` is a typical action a user would do. So consider to add a `console.log` with the submitted form data, when clicking submit.
* I would give the example index.html a meaningful title:
![Screenshot_20220302_100208](https://user-images.githubusercontent.com/42800119/156329502-d726b46d-8f13-4666-9f01-29ccdfcec323.png)

smbea commented:
@MaxTru I changed the html file name to viewer.html.
I added a `console.log` and also an `alert` just so the user has some visual feedback when submitting if they test the example.

MaxTru commented on custom-properties/viewer.html:
```diff
@@ -0,0 +1,89 @@
+<html>
+  <head>
+    <!--
```
consider to add
```html
  <title>Custom properties form-js example</title>
```

smbea replied:
✅

nikku reviewed (approved):
Looks good to me!
